File: SPO.Server/SPO.Domain/Entities/Payment.cs
using SPO.Domain.Entities.Base;

namespace SPO.Domain.Entities
{
    public class Payment : AuditableEntity<string>
    {
        public string PaymentMethod { get; set; } = string.Empty;
        public decimal Amount { get; set; }
        public string Status { get; set; } = string.Empty;
        public DateTimeOffset PaymentDate { get; set; }
        public string UserId { get; set; } = string.Empty;
    }
}

File: SPO.Client/app/types/payment.ts
export enum PaymentMethod {
  PayPal = "PAYPAL",
  CreditCard = "CREDIT_CARD",
  BankTransfer = "BANK_TRANSFER",
}

export interface Payment {
  paymentMethod: PaymentMethod;
  amount: number;
  status: string;
  userId: string;
}